Description: write the client pidfile before informing parent of success
 dhclient (by default) forks and waits for the child process to report its
 success/failure via a pipe. The child writes to the pipe before writing the
 pidfile, meaning the parent can exit before the pidfile is present, which
 breaks the attempt to kill dhclient process run in the initramfs to handle
 ip=dhcp.
Author: Michael Hudson-Doyle <michael.hudson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1819747
Forwarded: no
Last-Update: 2019-03-13
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/client/dhclient.c
+++ b/client/dhclient.c
@@ -4575,6 +4575,8 @@
 	if (dfd[0] == -1 || dfd[1] == -1)
 		return;
 
+	write_client_pid_file ();
+
 	/* Signal parent we started successfully. */
 	if (write(dfd[1], &buf, 1) != 1)
 		log_fatal("write to parent: %m");
@@ -4597,8 +4599,6 @@
 	(void) open("/dev/null", O_RDWR);
 	(void) open("/dev/null", O_RDWR);
 
-	write_client_pid_file ();
-
 	IGNORE_RET (chdir("/"));
 
 }
